try using GITHUB_TOKEN in CI and obtaining fewer 429s etc?

mlinksva opened the pull request.

mlinksva committed:
try passing token from workflow to rake build/htmlproofer

nschonni commented:
Probably need to do this in the `scripts` too

mlinksva committed:
with and run don't work together, maybe env is right?

mlinksva commented:
Yeah, will add elsewhere if/when this works.

nschonni commented on Rakefile:
```diff
@@ -4,9 +4,14 @@ require 'html-proofer'
+    hydra: { max_concurrency: 10 },
+    url_ignore: [%r{https://developer.github.com}, %r{https://docs.github.com}, %r{https://help.github.com}],
+    http_status_ignore: [429])
+  proofer.before_request do |request|
```
might want to check if the value is set, so the local script works if they haven't set this up. Maybe the empty header isn't an issue though

mlinksva committed:
actually access env variable

nschonni commented on Rakefile:
```diff
@@ -4,9 +4,14 @@ require 'html-proofer'
+    url_ignore: [%r{https://developer.github.com}, %r{https://docs.github.com}, %r{https://help.github.com}],
+    http_status_ignore: [429])
+  proofer.before_request do |request|
+    request.options[:headers]['Authorization'] = "Bearer #{GITHUB_TOKEN}" if request.base_url == "https://github.com"
```
I'm bad at ruby, but does it need to be something like `ENV["GITHUB_TOKEN"]`

mlinksva replied:
Me too, just noticed due to failing CI :)

mlinksva commented:
Mysteries:
* obviously not avoiding 429
* ~~htmlproofer config should ingore them~~ (updated htmlproofer config for new version)
* ~~Not sure why 2 workflows aren't running~~ (they only run if changes in `_data`, by design!)

nschonni commented:
Other approach might be to try getting it working on one of the other scripts with the native Typeous config, then use that for HTMLProofer https://github.com/gjtorikian/html-proofer/wiki/How-to-configure-Typhoeus

mlinksva committed:
https://github.com/gjtorikian/html-proofer/blob/3aa707371b28d87d01b429cec1f8d670bf16e6fa/UPGRADING.md?plain=1#L17

mlinksva committed:
use html proofer v4 as directives were renamed

mlinksva committed:
don't use token if not defined
dont' check external hashes due to fedramp site using them for js without including in html

mlinksva committed:
rubocop

mlinksva committed:
entity updates

mlinksva commented:
Not clear to me passing token is working, but other fixes here useful. Tests pass for 1st time in awhile.
